sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaTranslation.java

```diff
@@ -215,7 +215,19 @@ public static Schema schemaFromProto(SchemaApi.Schema protoSchema) {
     }
     builder.setOptions(optionsFromProto(protoSchema.getOptionsList()));
     Schema schema = builder.build();
-    schema.setEncodingPositions(encodingLocationMap);
+
+    Preconditions.checkState(encodingLocationMap.size() == schema.getFieldCount());
+    long dinstictEncodingPositions = encodingLocationMap.values().stream().distinct().count();
+    Preconditions.checkState(dinstictEncodingPositions <= schema.getFieldCount());
+    if (dinstictEncodingPositions < schema.getFieldCount() && schema.getFieldCount() > 0) {
+      // This means that encoding positions were not specified in the proto. Generally, we don't
+      // expect this to happen,
```

Comment:
Right now Python's row coder never sets encoding position, so Python pipelines that use SqlTransform will trigger this path